internal/failure_converter.go

```diff
@@ -174,7 +174,9 @@ func (dfc *DefaultFailureConverter) FailureToError(failure *failurepb.Failure) e
 	if failure == nil {
 		return nil
 	}
-
+	// Copy the original future to pass to the failureHolder
+	var orignalFailure failurepb.Failure
+	orignalFailure.XXX_Merge(failure)
```

Comment:
```suggestion
	originalFailure := proto.Clone(failure).(*failurepb.Failure)
```

This is the better proto cloning approach (`XXX_` methods should usually be avoided). Also, I fixed the spelling of `orignamFailure` (sic)